src/Microsoft.Data.SqlClient/add-ons/AzureKeyVaultProvider/SqlColumnEncryptionAzureKeyVaultProvider.cs

```diff
@@ -282,173 +276,69 @@ public override byte[] DecryptColumnEncryptionKey(string masterKeyPath, string e
+            return hash.Concat(signature).ToArray();
         }
 
         #endregion
 
         #region Private methods
 
-        /// <summary>
-        /// This function validates that the encryption algorithm is RSA_OAEP and if it is not,
-        /// then throws an exception
-        /// </summary>
-        /// <param name="encryptionAlgorithm">Asymmetric key encryption algorithm</param>
-        /// <param name="isSystemOp">is the operation a system operation</param>
-        private void ValidateEncryptionAlgorithm(ref string encryptionAlgorithm, bool isSystemOp)
-        {
-            // This validates that the encryption algorithm is RSA_OAEP
-            if (null == encryptionAlgorithm)
-            {
-                if (isSystemOp)
-                {
-                    throw new ArgumentNullException(Constants.AeParamEncryptionAlgorithm, Strings.NullAlgorithmInternal);
-                }
-                else
-                {
-                    throw new ArgumentNullException(Constants.AeParamEncryptionAlgorithm, Strings.NullAlgorithm);
-                }
-            }
-
-            // Transform to standard format (dash instead of underscore) to support both "RSA_OAEP" and "RSA-OAEP"
-            if (encryptionAlgorithm.Equals("RSA_OAEP", StringComparison.OrdinalIgnoreCase))
-            {
-                encryptionAlgorithm = JsonWebKeyEncryptionAlgorithm.RSAOAEP;
-            }
-
-            if (String.Equals(encryptionAlgorithm, JsonWebKeyEncryptionAlgorithm.RSAOAEP, StringComparison.OrdinalIgnoreCase) != true)
-            {
-                throw new ArgumentException(String.Format(CultureInfo.InvariantCulture, Strings.InvalidKeyAlgorithm,
-                                                            encryptionAlgorithm, "RSA_OAEP' or 'RSA-OAEP"), // For supporting both algorithm formats.
-                                            Constants.AeParamEncryptionAlgorithm);
-            }
-        }
-
-        private byte[] ComputeMasterKeyMetadataHash(string masterKeyPath, bool allowEnclaveComputations, bool isSystemOp)
-        {
-            // Validate the input parameters
-            ValidateNonEmptyAKVPath(masterKeyPath, isSystemOp);
-
-            // Validate whether the key is RSA one or not and then get the key size
-            GetAKVKeySize(masterKeyPath);
-
-            string masterkeyMetadata = ProviderName + masterKeyPath + allowEnclaveComputations;
-            byte[] masterkeyMetadataBytes = Encoding.Unicode.GetBytes(masterkeyMetadata.ToLowerInvariant());
-
-            // Compute hash
-            byte[] hash;
-            using (SHA256 sha256 = SHA256.Create())
-            {
-                sha256.TransformFinalBlock(masterkeyMetadataBytes, 0, masterkeyMetadataBytes.Length);
-                hash = sha256.Hash;
-            }
-            return hash;
-        }
 
         /// <summary>
         /// Checks if the Azure Key Vault key path is Empty or Null (and raises exception if they are).
```

Comment:
Nitpick: It looks like this does a bit more than simply check for null/empty now. Maybe update the summary and or method name to reflect this.